settings/src/main/java/org/odk/collect/settings/ODKAppSettingsImporter.kt

```diff
@@ -28,6 +28,10 @@ class ODKAppSettingsImporter(
     )
 
     fun fromJSON(json: String, project: Project.Saved): Boolean {
-        return settingsImporter.fromJSON(json, project)
+        return try {
+            settingsImporter.fromJSON(json, project)
+        } catch (e: Throwable) {
```

Comment:
I'm not sure if I agree with the issue (and it also seems more worried about camera code), but I think changing what we do with JSON errors makes sense, so I'll post in the issue and here. 

I think catching `JSONException` would be enough here as it's currently the only thing we'd expect that wouldn't be a code/build problem (like ProGuard problems) that would usually crash every time. And for testing that, I'd just add a test case with invalid JSON to `ODKAppSettingsImporterTest`.  We could alternatively keep the `catch` in `SettingsImporter` and just return `false` in that case. I'd be tempted to go with the former though as long term I suspect we'll  end up pulling `SettingsImporter` apart (discussed in https://github.com/getodk/collect/pull/4924).

Reply:
> I think catching JSONException would be enough here as it's currently the only thing we'd expect that wouldn't be a code/build problem (like ProGuard problems) that would usually crash every time.

Done.

> And for testing that, I'd just add a test case with invalid JSON to ODKAppSettingsImporterTest

I've extended the fix a bit and added catching `JsonParseException` see: https://github.com/getodk/collect/pull/5058/commits/70c84a67e8b90ddd7ef70977ece97ba930b48fbb

I've added a test in `ODKAppSettingsImporterTest` too but actually it doesn't test [JSONException](https://github.com/getodk/collect/pull/5058/commits/ffbaddb9f5b0caac0899f0ada4fa15c73ee6d7b1)
but that `JsonParseException`. I can't come up with any json that would not be caught in JsonSchemaSettingsValidator but cause  `JSONException` later.